src/event-handler/validate-event-name.ts

```diff
@@ -0,0 +1,54 @@
+import type { Logger } from "../create-logger.ts";
+import { emitterEventNames } from "../generated/webhook-names.ts";
+import type { EmitterWebhookEventName } from "../types.ts";
+
+type ValidateEventNameOptions =
+  | {
+      onUnknownEventName?: undefined | "throw";
+    }
+  | {
+      onUnknownEventName: "ignore";
+    }
+  | {
+      onUnknownEventName: "warn";
+      log?: Pick<Logger, "warn">;
+    };
+
+export function validateEventName<
+  O extends ValidateEventNameOptions = ValidateEventNameOptions,
+>(
+  eventName: EmitterWebhookEventName | (string & Record<never, never>),
+  options: O = {} as O,
+): asserts eventName is O extends { onUnknownEventName: "throw" }
```

Comment:
It's best to use descriptive names as much as possible.

TOptions would suit fine.

```suggestion
  TOptions extends ValidateEventNameOptions = ValidateEventNameOptions,
>(
  eventName: EmitterWebhookEventName | (string & Record<never, never>),
  options: TOptions = {} as TOptions,
): asserts eventName is TOptions extends { onUnknownEventName: "throw" }
```